cmd/helm/upgrade.go

```diff
@@ -91,10 +92,16 @@ func newUpgradeCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 
 			if client.Install {
 				// If a release does not exist, install it. If another error occurs during
-				// the check, ignore the error and continue with the upgrade.
+				// the check, ignore the error (unless it indicates the cluster is not reachable)
+				// and continue with the upgrade.
 				histClient := action.NewHistory(cfg)
 				histClient.Max = 1
-				if _, err := histClient.Run(args[0]); err == driver.ErrReleaseNotFound {
+
+				_, err := histClient.Run(args[0])
```

Comment:
Is it possible to capture other non-nil errors as well? e.g.

```
if _, err := histClient.Run(args[0]); err == driver.ErrReleaseNotFound {
    ...
} else if err != nil {
    return err
}
```

That way we don't have to refactor this in the future if we introduce further non-nil errors

Reply:
Good point!  Done.